src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -173,7 +174,7 @@ private void BlockManagerOnSystemContractInvoked(object _, InvocationContext con
                     return;
                 }
 
-                var sender = keygen.GetSenderByPublicKey(context.Receipt.RecoverPublicKey());
+                var sender = keygen.GetSenderByPublicKey(context.Receipt.RecoverPublicKey(HardforkHeights.IsHardfork_8Active(context.Snapshot.Blocks.GetTotalBlockHeight())));
```

Comment:
Should it be context.Receipt.Block to check hardfork8 ?

Reply:
fixed